pypeit/edgetrace.py

```diff
@@ -129,6 +130,102 @@
+                 'pad': dict(otype=int,
+                             descr='Integer number of pixels to consider beyond the slit edges.'),
+                 'nslits': dict(otype=int, descr='Number of slits.'),
+                 'left_orig': dict(otype=np.ndarray, atype=float,
+                                   descr='The original spatial coordinates (pixel indices) of all '
+                                         'left edges, one per slit.  Shape is Nspec by Nslits.'),
+                 'right_orig': dict(otype=np.ndarray, atype=float,
+                                    descr='The original spatial coordinates (pixel indices) of all '
+                                          'right edges, one per slit.  Shape is Nspec by Nslits.'),
+                 'left': dict(otype=np.ndarray, atype=float,
+                              descr='Spatial coordinates (pixel indices) of all left edges, one '
+                                    'per slit.  Shape is Nspec by Nslits.'),
+                 'right': dict(otype=np.ndarray, atype=float,
+                              descr='Spatial coordinates (pixel indices) of all right edges, one '
+                                    'per slit.  Shape is Nspec by Nslits.'),
+                 'center': dict(otype=np.ndarray, atype=float,
+                               descr='Spatial coordinates of the slit centers.  Shape is Nspec '
+                                     'by Nslits.'),
+                 'mask': dict(otype=np.ndarray, atype=bool,
+                              descr='Bad-slit mask (good slits are False).  Shape is Nslits.'),
+                 'specmin': dict(otype=np.ndarray, atype=float,
+                                 descr='Minimum spectral position allowed for each slit/order.  '
+                                       'Shape is Nslits.'),
+                 'specmax': dict(otype=np.ndarray, atype=float,
+                                 descr='Maximum spectral position allowed for each slit/order.  '
+                                       'Shape is Nslits.')} 
+    """Provides the class data model."""
+    # NOTE: The docstring above is for the ``datamodel`` attribute.
+
+    def __init__(self, left, right, nspat=None, spectrograph=None, left_orig=None, right_orig=None,
+                 mask=None, specmin=None, specmax=None, binspec=1, binspat=1, pad=0):
+        args, _, _, values = inspect.getargvalues(inspect.currentframe())
+        super(SlitTraceSet, self).__init__({k: values[k] for k in args[1:]})
+
+    def _validate(self):
+        """
+        Validate the slit traces.
+        """
+        if self.left.shape != self.right.shape:
+            raise ValueError('Input left and right traces should have the same shape.')
+        self.nspec, self.nslits = self.left.shape
+        self.center = (self.left+self.right)/2
+        
+        if self.nspat is None:
+            self.nspat = np.amax(np.append(self.left, self.right))
+        if self.spectrograph is None:
```

Comment:
I haven't integrated this into the rest of the code yet, but if the spectrograph name was needed in `tslits_dict` then you're right and it will be needed here.  My next PR is replacing the `tslits_dict` with the `SlitTraceSet` object, so I'll make the change then.